src/components/organisms/ImageUpload/index.js

```diff
@@ -0,0 +1,223 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-fetches'
+
+import ImageDialog from '_molecules/image-dialog'
+import WebcamCapture from '_molecules/webcam-capture'
+
+import styles from './styles.css'
+
+const mapDispatchToProps = (http, dispatch) => ({
+  uploadImage: dispatch(http.post('upload/avatar'), {
+    headers: { 'content-type': 'multipart/form-data' },
+  }),
+})
+
+class ImageUpload extends Component {
+  state = {
+    selectedFile: null,
+    notAcceptedFileSize: null,
+    notAcceptedFileDimensions: null,
+    notAcceptedFileExtension: null,
+    showImagePreview: false,
+    imageUpload: '',
+  }
+
+  onImageSelectedOrCaptured = imageSrc => {
+    if (this.props.withPreview) {
+      this.setImagePreviewState(imageSrc)
+    } else {
+      this.setState({ selectedFile: imageSrc })
+    }
+  }
+
+  setImagePreviewState = imageFile => {
+    this.setState({ selectedFile: imageFile, showImagePreview: true })
+  }
+
+  uploadImage = () => {
+    const { selectedFile } = this.state
+    this.props.uploadImage({ file: selectedFile }).then(({ data, error }) => {
+      this.setState(prevState => ({
```

Comment:
you don't need to use setState as a function when you are not using the prevState.